vm/actor/src/builtin/system/mod.rs

```diff
@@ -3,30 +3,41 @@
 
 use crate::{check_empty_params, SYSTEM_ACTOR_ADDR};
 
+use encoding::Cbor;
 use ipld_blockstore::BlockStore;
 use num_derive::FromPrimitive;
 use num_traits::FromPrimitive;
 use runtime::{ActorCode, Runtime};
-use vm::{ActorError, ExitCode, MethodNum, Serialized, METHOD_CONSTRUCTOR};
+use serde::{Deserialize, Serialize};
+use vm::{actor_error, ActorError, ExitCode, MethodNum, Serialized, METHOD_CONSTRUCTOR};
 
-/// Init actor methods available
+// * Updated to specs-actors commit: b8a3a6ff7b15ac01f0534c47059e1c81652a61f0 (v0.9.1)
+
+/// System actor methods.
 #[derive(FromPrimitive)]
 #[repr(u64)]
 pub enum Method {
     Constructor = METHOD_CONSTRUCTOR,
 }
 
-/// Init actor
+/// System actor state.
+#[derive(Default, Deserialize, Serialize)]
+#[serde(transparent)]
+pub struct State([(); 0]);
```

Comment:
This is just for me but why us State([(); 0]) instead of something like State{}

Reply:
Because that would serialize as an empty map, when it needs to serialize as an empty array (different bytes in Cbor)